src/python/dxpy/bindings/apollo/ValidateJSONbySchema.py

```diff
@@ -0,0 +1,165 @@
+from ...exceptions import err_exit
+
+
+class JSONValidator(object):
+    """
+    JSON validator class to validate a JSON against a schema.
+
+    The schema is a dictionary with the following structure:
+
+    {
+        "key1": {
+            "type": dict,
+            "properties": {
+                "name": {"type": str, "required": True},
+                "attribute": {"type": str}
+            }
+        },
+        "key2": {
+            "type": dict,
+            "properties": {
+                "xyz": {"type": str},
+                "abc": {"type": list}
+            },
+            "conflicting_keys": [["xyz", "abc"]]
+        },
+        "list_key_example": {
+            "type": list,
+            "items": {
+                "type": dict,
+                "properties": {
+                    "nested_key1": {"type": str, "required": True},
+                    "nested_key2": {"type": str}
+                }
+            }
+        },
+        "id": {
+            "type": list,
+        },
+        "conflicting_keys": [["key1", "key2"]],
+        "dependent_conditional_keys": {
+            "list_key_example": ["key1", "key2"]
+        }
+    }
+
+    Key types can be defined via the "type" key
+
+    Required keys can be defined via the "required" key
+
+    Conflicting keys can be defined via the "conflicting_keys" key, where the value is a list of lists of keys that are mutually exclusive.
+    Conflicting keys can be defined at the level of the schema for the entire JSON or at key-level.
+    In the example above, within "key2" the keys "xyz" and "abc" are mutually exclusive and cannot be present together.
+    Similarly, at the level of the whole JSON, "key1" and "key2" are mutually exclusive and cannot be present together.
+
+    Conditional key combinations can be defined via the dependent_conditional_keys at the level of the schema. These should be defined as a dict where the value is a list.
+    In the example above, dependent_conditional_keys is defined as {"list_key_example": ["key1", "key2"]}, which means whenever list_key_example is present, either "key1"
+    or "key2" has to be also present.
+
+    Currently the maximum level of nestedness supported for validation is as deep as shown in the example above via the "list_key_example" key.
+
+    This is an example of a valid JSON that satisfies all the conditions defined in the schema above:
+
+    {
+        "key1": {
+            "name": "test",
+        },
+        "list_key_example": [
+            {"nested_key1": "1", "nested_key2": "2"},
+            {"nested_key1": "3", "nested_key2": "4"},
+            {"nested_key1": "5"}
+        ],
+        "id": ["id_1", "id_2"]
+    }
+
+    """
+
+    def __init__(self, schema):
+        self.schema = schema
```

Comment:
We need to check for `None` schema as downstream none-None is assumed.

Reply:
Addressed here: https://github.com/dnanexus/dx-toolkit/pull/1157/commits/9b336e65ec0c508aaf1619ac09d98e3208fb42a3